### PR TITLE
[frontend] indicator generation Text displayed false (#6845)

### LINKED ISSUE #6845: [Indicator generation] Text displayed false
## Description

The text displayed for generating indicators from observables or observables from indicators is the same (so wrong in one case ;) )

## Environment

OCTI 6.0.10

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Go to Observations > Observables
2. Select two observables
3. Click on "Indicators/Observables generation" in the bottom right corner
4. Read the text displayed in the orange square

## Expected Output

I expect "This action will generate indicators from the selected observables."

## Additional information

As the text displayed here is the same as when observables are generated from incidators, I wonder whether the same variable is used in the code. Be careful not to modify the text on both sides and end up with "This action will generate indicators from the selected observables." when generating observables from indicators ;)

## Screenshots

Screenshot 2024-04-30 085148

### PR DESCRIPTION
**Pull request by ValentinBouzinFiligran**
### Proposed changes

* promote drawer on a new file
* new translation depending of type added

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/6845
*

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by SouadHadjiat — CHANGES_REQUESTED** [review]
There is an issue when opening a report -> entities -> select indicators only -> click on toolbar "promote" action => nothing happens, the drawer doesn't open.

File: opencti-platform/opencti-front/src/private/components/data/drawers/PromoteDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+type DrawerContainerPropsType = {
+  isOpen: boolean;
+  observablesFiltered: boolean;
+  onClose: () => void;
+  onSubmit: () => void;
+  isIndicator: boolean;
+};
+
+const PromoteDrawer = ({ isOpen, onClose, observablesFiltered, onSubmit, isIndicator }: DrawerContainerPropsType) => {
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/data/drawers/PromoteDrawer.tsx]: I don't really understand the parameters observablesFiltered and isIndicator. What values can they have ? Can we have observablesFiltered and isIndicator true ? so we have observables and indicators to promote at the same time ?

**Review by SouadHadjiat — APPROVED** [review]
tested locally

File: opencti-platform/opencti-front/src/private/components/data/drawers/PromoteDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    marginTop: 20,
+    textAlign: 'right',
+  },
+  button: {
+    marginLeft: theme.spacing(2),
+  },
+  container: {
+    padding: '10px 20px 20px 20px',
+  },
+}));
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/data/drawers/PromoteDrawer.tsx]: `makeStyles` is deprecated, use properties `sx` and `style` directly in JSX instead

File: opencti-platform/opencti-front/src/private/components/data/drawers/PromoteDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  const classes = useStyles();
+  const { t_i18n } = useFormatter();
+
+  return (
+    <Drawer
+      title={t_i18n('Observables and indicators conversion')}
+      onClose={onClose}
+      open={isOpen}
+    >
+      <>
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/data/drawers/PromoteDrawer.tsx]: No need for this empty fragment as there is only one child

**Review by lndrtrbn — CHANGES_REQUESTED** [review]

File: opencti-platform/opencti-front/src/private/components/data/drawers/PromoteDrawer.tsx (hunk 2 above)
Comment (lndrtrbn) [reply opencti-platform/opencti-front/src/private/components/data/drawers/PromoteDrawer.tsx]: You can have a look at `WidgetTimeline.tsx` if you need an example

**Review by lndrtrbn — APPROVED** [review]

**Pull request merged**